### PR TITLE
🐛 Fix isDbError()-guard does not work

### LINKED ISSUE #12400: isDbError() does not work
### Astro Info

```block
Astro                    v4.16.10
Node                     v22.6.0
System                   macOS (arm64)
Package Manager          pnpm
Output                   server
Adapter                  @astrojs/vercel/serverless
Integrations             @astrojs/tailwind
                         @astrojs/svelte
                         astro:db
                         @astrojs/db/file-url
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

`isDbError(err)` is false, even though the error is a LibsqlError `SQLITE_CONSTRAINT_UNIQUE` thrown from `db.insert` function in an Astro action.

```ts
      try {
        const user = await db.insert(User).values({ name }).returning().get(); // * throws LibsqlError `SQLITE_CONSTRAINT_UNIQUE` when user with name already exists.
        return user;
      } catch (err) {
        if (isDbError(err)) {
          console.warn("LibsqlError thrown:", err); // ! this never get's called, since `isDbError(err)` doesn't seem to work
        } else {
          console.error("Error thrown:", err);
        }
      }
```

Notice: the StackBlitz repro works only locally, since StackBlitz doesn't support Astro.clientAddress: https://github.com/withastro/astro/issues/6451#issuecomment-1488618493

### What's the expected result?

I would expect to be able to identify Libsql errors thrown from inserting into db with astro.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-ofpscu-cwqc8l?file=src%2Factions%2Findex.ts

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by abegehr**
Fixes #12400

## Changes

- fixes https://github.com/withastro/astro/issues/12400
- manually check LibsqlError type through ts-guard-function
- `instanceof LibsqlError` was not sufficient, possibly because of this: https://stackoverflow.com/a/58050942/499113

## Testing

I tested in a separate project ([here](https://github.com/withastro/astro/blob/main/CONTRIBUTING.md#development)
 option 3), since atm there is no example project with db configured

## Docs

this is a bug fix, therefore no docs update needed

### CODE REVIEW

File: packages/db/src/runtime/virtual.ts [hunk 1]
```diff
@@ -22,7 +22,18 @@ function createColumn<S extends string, T extends Record<string, unknown>>(type:
...
+	const value = err as LibsqlError;
+	return (
+		// Error
+		typeof value.name === 'string' &&
+		typeof value.message === 'string' &&
+		(value.stack === undefined || typeof value.stack === 'string') &&
+		// LibsqlError
+		typeof value.code === 'string' &&
+		(value.rawCode === undefined || typeof value.rawCode === 'number')
+	);
```
Comment (bluwy) [inline packages/db/src/runtime/virtual.ts:36]: It looks like the libsql errors also have the `libsqlError: true` property which perhaps should be sufficient enough for the checks? Here was the logs:

<details>
<summary>open logs</summary>

```
Error thrown: Error: UNIQUE constraint failed: User.name
    at Object.next (/Users/bjorn/Work/repros/astro-issue-12400-repro/node_modules/.pnpm/libsql@0.4.7/node_modules/libsql/index.js:335:20)
    at Statement.all (/Users/bjorn/Work/repros/astro-issue-12400-repro/node_modules/.pnpm/libsql@0.4.7/node_modules/libsql/index.js:360:16)
    at executeStmt (file:///Users/bjorn/Work/repros/astro-issue-12400-repro/node_modules/.pnpm/@libsql+client@0.14.0/node_modules/@libsql/client/lib-esm/sqlite3.js:261:34)
    at Sqlite3Client.execute (file:///Users/bjorn/Work/repros/astro-issue-12400-repro/node_modules/.pnpm/@libsql+client@0.14.0/node_modules/@libsql/client/lib-esm/sqlite3.js:79:16)
    at LibSQLPreparedQuery.values (file:///Users/bjorn/Work/repros/astro-issue-12400-repro/node_modules/.pnpm/drizzle-orm@0.31.4_@libsql+client@0.14.0/node_modules/drizzle-orm/libsql/session.js:170:59)
    at LibSQLPreparedQuery.get (file:///Users/bjorn/Work/repros/astro-issue-12400-repro/node_modules/.pnpm/drizzle-orm@0.31.4_@libsql+client@0.14.0/node_modules/drizzle-orm/libsql/session.js:143:29)
    at QueryPromise.get (file:///Users/bjorn/Work/repros/astro-issue-12400-repro/node_modules/.pnpm/drizzle-orm@0.31.4_@libsql+client@0.14.0/node_modules/drizzle-orm/sqlite-core/query-builders/insert.js:148:28)
    at handler (/Users/bjorn/Work/repros/astro-issue-12400-repro/src/actions/index.ts:15:117)
    at eval (/Users/bjorn/Work/repros/astro-issue-12400-repro/node_modules/.pnpm/astro@4.16.10_typescript@5.6.3/node_modules/astro/dist/actions/runtime/virtual/server.js:68:18)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  rawCode: 2067,
  code: 'SQLITE_CONSTRAINT_UNIQUE',
  libsqlError: true
}
```
</details>

Perhaps we can simplify the check here as: 
```suggestion
	return err instanceof LibsqlError || (err instanceof Error && (err as any).libsqlError === true)
```

**Review by bluwy — COMMENTED** [review]
It looks like there's other places where we also do a plain `instanceof LibsqlError` that may hit this issue, but that could be tackled separately.

File: packages/db/src/runtime/virtual.ts (hunk 1 above)
Comment (abegehr) [reply packages/db/src/runtime/virtual.ts:36]: yes, that should work, only reservation I would have is that libsqlError is not included on LibsqlError type, so need to cast to any

File: packages/db/test/test-utils.js [hunk 2]
```diff
@@ -142,7 +142,7 @@ function createRemoteDbServer() {
 					JSON.stringify({
 						success: false,
 						error: {
-							code: e instanceof LibsqlError ? e.code : 'SQLITE_QUERY_FAILED',
+							code: isDbError(e) ? e.code : 'SQLITE_QUERY_FAILED',
```
Comment (bluwy) [inline packages/db/test/test-utils.js:145]: isDbError isn't imported

File: packages/db/src/core/cli/commands/execute/index.ts [hunk 3]
```diff
@@ -16,6 +15,7 @@ import {
 import { bundleFile, importBundledFile } from '../../../load-file.js';
 import type { DBConfig } from '../../../types.js';
 import { getManagedRemoteToken } from '../../../utils.js';
+import { isDbError } from '../../../../runtime/virtual.js';
```
Comment (bluwy) [inline packages/db/src/core/cli/commands/execute/index.ts:18]: I don't know if this causes the CI fail, but I _think_ we may need to copy `isDbError` into `../../../utils.js` so there's two implementations of it for now. The `virtual.js` is only for the `astro:db` virtual module and we don't want to mix it with the non-virtual code.

File: .changeset/breezy-radios-grab.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/db': patch
+---
+
+fix isDbError()-guard for LibsqlError
```
Comment (bluwy) [inline .changeset/breezy-radios-grab.md:5]:
```suggestion
Fixes `isDbError()` guard for `LibsqlError`
```

**Review by bluwy — APPROVED** [review]

**Review by ematipico — APPROVED** [review]

**Pull request merged**